gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java

```diff
@@ -232,8 +244,12 @@ private Map<String, Object> prepareEmailParams(User user, Client client, Integer
             claims.put(Claims.aud, client.getId());
         }
 
-        if (client != null) {
-            queryParams.add("client_id", encodeURIComponent(client.getClientId()));
+        if (client != null && !queryParams.contains(CLIENT_ID)) {
+            queryParams.add(CLIENT_ID, encodeURIComponent(client.getClientId()));
+        }
+
+        if(queryParams.contains(REDIRECT_URI)) {
+            encodeAndReplaceRedirectURIQueryParam(queryParams);
```

Comment:
**thought**: I'm not sure that this is the fix we should apply :thinking: I wondering instead with we provide a redirect_uri parameter to the validationRegistration endpoint as it not used... wdyt ?

Reply:
I just check, and for the forgotPassword for example, we are not providing the parameters coming from the request. That make sens IMO, I wondering why it has been done in another way for registrationValidation... I think we should do the same as forgotPassword. wdyt ?

Reply:
code updated